src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/SynthesizedSymbols/AnonymousType.DelegateTemplateSymbol.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed partial class AnonymousTypeManager
+    {
+        internal sealed class AnonymousDelegateTemplateSymbol : AnonymousTypeOrDelegateTemplateSymbol
+        {
+            private readonly ImmutableArray<Symbol> _members;
+
+            internal readonly bool IsParameterizedDelegateType;
+
+            internal AnonymousDelegateTemplateSymbol(
+                AnonymousTypeManager manager,
+                string name,
+                TypeSymbol objectType,
+                TypeSymbol intPtrType,
+                TypeSymbol? voidReturnTypeOpt,
+                int parameterCount,
+                RefKindVector refKinds)
+                : base(manager, Location.None)
```

Comment:
>Location.None [](http://example.com/codeflow?start=32&length=13)

Do we never have ```AnonymousTypeDescriptor``` for delegates coming through here?

Reply:
The `location` parameter is not needed for this constructor since `NameAndIndex` is set explicitly below. Added a comment.